main.py

```diff
@@ -71,18 +76,36 @@ def main(argv: Sequence[str] | None = None) -> int:
 
     os.chdir(args.pwd)
 
-    # generic/standalone tools
+    # generic/standalone tools that do not care about devenv configuration
     if args.command == "pin-gha":
         return pin_gha.main(remainder)
-    # Future home of bootstrap-sentry, and then bootstrap-*.
 
-    # the remaining tools are repo-specific
+    config = configparser.ConfigParser()
+    config_path = f"{config_root}/config.ini"
+
+    if not os.path.exists(config_path):
+        coderoot = os.path.expanduser(
+            input("please enter the root directory where you want to work in [~/dev]: ")
+        ) or os.path.expanduser("~/dev")
+        os.makedirs(coderoot, exist_ok=True)
+
+        config["devenv"] = {"coderoot": coderoot}
+
+        os.makedirs(config_root, exist_ok=True)
+        with open(config_path, "w") as f:
+            config.write(f)
+        print(f"If you made a mistake, you can edit {config_path}.")
 
-    # TODO: read a well-known json for preferences
-    coderoot = "dev"
+    config.read(config_path)
+    coderoot = config["devenv"]["coderoot"]
+
+    if args.command == "bootstrap":
+        return bootstrap.main(coderoot, remainder)
+
+    # the remaining tools are repo-specific
     if not args.nocoderoot and not args.pwd.startswith(os.path.expanduser(f"~/{coderoot}")):
         print(
-            f"You aren't in your code root (~/{coderoot})!"
+            f"You aren't in your code root (~/{coderoot})! "
```

Comment:
Is there a reason for the extra space here?

Reply:
Yep, it's concatenated with the string below.